src/checker.cpp

```diff
@@ -4310,7 +4310,7 @@ gb_internal bool correct_single_type_alias(CheckerContext *c, Entity *e) {
 
 gb_internal bool correct_type_alias_in_scope_backwards(CheckerContext *c, Scope *s) {
 	bool correction = false;
-	for (u32 n = s->elements.count, i = n-1; i < n; i--) {
+	for (u32 n = s->elements.capacity, i = n-1; i < n; i--) {
```

Comment:
This change is unrelated to the LLVM 18 PR but an existing bugfix.

To explain, this array is sparse/has holes so having 3 items doesn't mean it has one at index 1, 2 and 3. It might be at 1, 5, and 8 for example.

This means the entire capacity has to be looped and checked for valid indices inside the body.